### PR TITLE
HBHE M3 = fix M3 

### PR DESCRIPTION
**Pull request by mariadalfonso**
reset the time slew correction for siPM

Was already done in #17094, bypassed by the #17563

Validation slides
http://dalfonso.web.cern.ch/dalfonso/M3SwitchOffTimeSlewfix.pdf

@fwyzard @deguio

### CODE REVIEW

**Comment by arunhep** [comment]
please test

File: CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc [hunk 1]
```diff
@@ -8,22 +8,25 @@ static const double cap = 6.0;
 static const double tspar0[2] = {15.5, 12.2999};
 static const double tspar1[2] = {-3.2,-2.19142};
 static const double tspar2[2] = {32, 0};
-static const double tspar0_siPM[2] = {3., 3.}; // 3ns delay for MC and DATA, recheck later for data
+static const double tspar0_siPM[2] = {0., 0.}; // 2ns delay for MC and DATA, recheck later for data
```
Comment (kpedro88) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: Is there a reason the comment says 2ns, but the values are 0?

Comment (mariadalfonso) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: @ kpedro.
I checked the M3 scale with the additional offset of 0ns or 2ns and doens't change.
The point of this PR is to remove the energy evolution of the correction.

With M2 time I measure now 2ns as delay of the MC digi. Before the addition of the Photostatistics, it was 3 ns. Given that the M2 time change while the M2 evolve and that the HPD M2 show the two population, I prefer to not to use this extra delay and I kept only the 6ns (the delay used in the digitization).

Comment (fwyzard) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: ehm, for somebody who is not an HCAL expert, this comment is completely impossible to understand...

Comment (fwyzard) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: doesn't this partially revert #17094 ?

**Comment by franzoni** [comment]
https://github.com/cms-sw/cmssw/pull/17984#issuecomment-288107315

File: CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc (hunk 1 above)
Comment (mariadalfonso) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: @fwyzard reching, the enum "source==data/MC/InputPars" is different in the various if statement

Comment (mariadalfonso) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: @slava77 in this if,  InputPars=3 and tspar2_siPM[3-1]  but the tspar2_siPM is an array with only two components :(

Comment (slava77) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: @mariadalfonso 
I didn't check this yet, I'm guessing your response is to @fwyzard 

Indeed, the current change for InputPars=3 is buggy

**Comment by mariadalfonso** [comment]
@fwyzard @slava77 
array fixed, slides in the PR description updated

**Comment by slava77** [comment]
assign reconstruction

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
hi @slava77 - as discussed in the orp - i will merge as its confirmed this PR improves things. any further feedback is still appreciated. thx

**Comment by slava77** [comment]
@davidlange6 
fine to merge based on earlier tests

jenkins tests here will probably be useless, because apparently they didn't start before the merge

also https://cmssdt.cern.ch/jenkins/job/ib-any-integration/ is not accessible

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17983	5333456
- post-factum signoff
- jenkins tests pass and (fractional) phase-1 HCAL workflows show changes only in method-3 values as expected from http://dalfonso.web.cern.ch/dalfonso/M3SwitchOffTimeSlewfix.pdf
e.g. the following from 10824 shows M3 much closer to the diagonal
wf10824_m3_m0_he

**Pull request merged**